rust/src/dhcp/dhcp.rs

```diff
@@ -19,10 +19,11 @@ use crate::applayer::{self, *};
 use crate::core::{ALPROTO_UNKNOWN, AppProto, IPPROTO_UDP};
 use crate::dhcp::parser::*;
 use crate::flow::Flow;
+use crate::sys::AppProtoEnum;
 use std;
 use std::ffi::CString;
 
-pub(super) static mut ALPROTO_DHCP: AppProto = ALPROTO_UNKNOWN;
+pub(super) static ALPROTO_DHCP: AppProto = AppProtoEnum::ALPROTO_DHCP as AppProto;
```

Comment:
Did you see https://github.com/OISF/suricata/pull/12420 @jasonish ?

Reply:
Yeah, but its not merged right?  This is an old commit, from October. Is it wrong given current master?

Reply:
This is not merged, but this is kind of concurrent to this change.

> This is an old commit, from October.

PR is from last week

> Is it wrong given current master?

I remember doing such a PR once, and then rejected the ticket as we want dynamic app-layer protocols, and this is not what is done in a rust enum...

See https://redmine.openinfosecfoundation.org/issues/3524